package/base-files/files/bin/config_generate

```diff
@@ -250,9 +251,12 @@ generate_static_system() {
 		add_list system.ntp.server='3.openwrt.pool.ntp.org'
 	EOF
 
+	local hostname
```

Comment:
Here, you would delete the `set system.@system[-1].hostname='OpenWrt'` in the uci batch above and then add

```
local labelmac=$(get_mac_label)
if [ -n "$labelmac" ]; then
   set system.@system[-1].hostname="OpenWrt_$(macaddr_geteui $labelmac)"
else
   set system.@system[-1].hostname=OpenWrt
fi
```
_before_ the "if json_is_a ..." block.

The remaining changes in this patch can be dropped.